scene/resources/material.cpp

```diff
@@ -1036,6 +1039,9 @@ void SpatialMaterial::_update_shader() {
 			case BLEND_MODE_MUL: {
 				code += "\tvec3 detail = mix(ALBEDO.rgb,ALBEDO.rgb*detail_tex.rgb,detail_tex.a);\n";
 			} break;
+			case BLEND_MODE_PMALPHA: {
+				// Not used for detail blend modes, but strict compiler rules cause errors if this case is not handled.
```

Comment:
If this situation shouldn't be possible, maybe we can print a warning for this case, so it will get noticed if it ever goes wrong.

Reply:
Its not that it isn't possible, its just that the compiler insists that every case be represented in a switch statement. So this line of code tells the reader that we intentionally choose not to insert any code in the premul alpha case

Reply:
Technically, you could use premultiplied alpha for as a detail map blend mode, but it probably doesn't make much sense and wouldn't be used often.